tests/test_cache.py

```diff
@@ -195,90 +200,164 @@ def test_cache_using_string(self):
         TestCache.test_value = 100
         self.assertEqual(self.get_cache_expire_string_year(), 100)
 
-    @Cache(key="test_cache_1", expire=timedelta(seconds=2))
+    def _test_empty_cache_helper(self, cache_data, empty_value, cache_empty_results):
+        if cache_empty_results:
+            self.clear_cache_empty()
+            self.assertEqual(self.get_cache_empty_results(empty_value), empty_value)
+            self.assertEqual(self.get_cache_empty_results(cache_data), empty_value)
+            with freeze_time(datetime.utcnow() + timedelta(seconds=2)):
+                self.assertEqual(self.get_cache_empty_results(cache_data), cache_data)
+            self.assertEqual(self.get_cache_empty_results(empty_value), cache_data)
+            with freeze_time(datetime.utcnow() + timedelta(seconds=4)):
+                self.assertEqual(self.get_cache_empty_results(empty_value), empty_value)
+        else:
+            self.clear_cache_empty_false()
+            self.assertEqual(self.get_dont_cache_empty_results(empty_value), empty_value)
+            self.assertEqual(self.get_dont_cache_empty_results(cache_data), cache_data)
+            TestCache.test_dict = {}

```

Comment:
is this related tp the test?

Reply:
sorry, my mistake, had to remove that while pushing the code